test/specs/reference/tests/04.paging.js

```diff
@@ -139,17 +138,60 @@ exports.execute = function (options) {
                         expect(tuples[i]._data["value x"]).toBeLessThan(tuples[i+1]._data["value x"]);
                     }
                 });
+
+                // limit was changed after paging back
+                it("read with an increased limit should return a Page object that is defined and doesn't have a previous page.", function(done) {
```

Comment:
Although these are testing the scenario that you led to finding the bug; they don't look like unit tests.
You could simply test this by directly calling read on a reference with before options, when
- Number of returned entities are less than limit.
- It's the beginning of the set.